### PR TITLE
Reworks the fishing minigame into a game screen object from a TGUI interface

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This PR removes the tgui interface previously used for the fishing minigame and recreates it in dream maker. It's still server side, but without the extra latency from the tgui interface. I'm also aiming to make the bait easier to control and therefore the game.

The chances of jumping for the fish have been adjusted slightly too, since it likes to do it better now.

here's an old screenshot:
le fishe

You can find videos of the feature in this [comment](https://github.com/tgstation/tgstation/pull/78052#issuecomment-1703924726).

## Why It's Good For The Game
Everyone and their mother has complained that the fishing minigame UI is laggy, to the point it's become the very core issue of the feature, which invalidates the contents build onto it from being entertaining.
Alas, I have no idea how to fix that while it's still a js/react UI running on a framework deep and beyond my humble lack of proficency, implying it can even be done as of now. Therefore I'm recreating it in Dream Maker.

## Changelog

:cl: Ghommie (Thanks Sealed101)
refactor: Reworked the fishing minigame into a game screen object from a TGUI interface
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
i was asked to record the new fishing
<details><summary>fishing videos (no gachi bgm this time)</summary>
<p>

https://github.com/tgstation/tgstation/assets/75863639/7b8bec5c-1be3-47a0-bb54-164d9a1a62c5

https://github.com/tgstation/tgstation/assets/75863639/c31407d0-478d-496f-8a19-6b697cfcef41

https://github.com/tgstation/tgstation/assets/75863639/98c08557-de14-42f3-b313-ed2d0c7a97f1

https://github.com/tgstation/tgstation/assets/75863639/30d98c1d-20fe-42ca-a3a0-ba371c7cdf4f

https://github.com/tgstation/tgstation/assets/75863639/bf580b4a-f1f0-4ace-88c8-15b73af4cfe7

</p>
</details>

**Comment by Ghommie** [comment]
Oh, I've got to fix the background for lavaland.

**Comment by Mothblocks** [comment]
i dont believe that this is faster and more responsive (especially with average player ping) than what the tgui interface could be. if its laggy now i would rather we look into that because i call bs

**Comment by Ghommie** [comment]
I've already looked into it and couldn't make head or tail on what makes it (and TGUI interfaces in general) choppier than the game screen itself. On its own, without a connection to a server, the UI is undeniably smoother and more responsive than this will ever be, but as long as a connection is there, it sucks. Perhaps having a better computer may mitigate it, and definitely a better connection will do a bunch.

The skepticism is acceptable and you might as well be right, but it wouldn't hurt to have it testmerge to get some actual feedback to see if thats the case (though I'd have to temporarily comment/revert some improvements for the sake of a better evaluation).

**Comment by Ghommie** [comment]
#78105 comes first.

File: code/modules/fishing/fishing_minigame.dm [hunk 1]
```diff
@@ -211,57 +318,238 @@
...
+	hud_bait.icon_state = "bait"
+	if(challenge.bait_pixel_height != MINIGAME_BAIT_HEIGHT)
+		var/height_percent_diff = challenge.bait_pixel_height/MINIGAME_BAIT_HEIGHT
+		hud_bait.transform = hud_bait.transform.Scale(1, height_percent_diff)
+		hud_bait.pixel_z = -icon_height * (1 - height_percent_diff) * 0.5
+	hud_bait.vis_flags = VIS_INHERIT_ID
+	vis_contents += hud_bait
+	hud_fish = new
+	hud_fish.icon = icon
+	hud_fish.icon_state = "fish"
```
Comment (AnturK) [inline code/modules/fishing/fishing_minigame.dm:538]: could move all these initializers to screen object subtypes

File: code/controllers/subsystem/processing/fishing.dm [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * So far, used only by the fishing minigame. Feel free to rename it to something like veryfastprocess
+ * if you need one that fires 10 times a second
+ */
+PROCESSING_SUBSYSTEM_DEF(fishing)
+	name = "Fishing"
+	wait = 0.1 SECONDS
+	can_fire = FALSE
```
Comment (AnturK) [inline code/controllers/subsystem/processing/fishing.dm:8]: Some debugging leftover, i assume.

**Comment by Ghommie** [comment]
I suppose the players liked this one better for it to be merged. Interesting.

**Pull request merged**